fetch_models.js

```diff
@@ -0,0 +1,163 @@
+    },
+    {
+      url: "microsoft/stable-diffusion-v1.5-webnn/resolve/main/sd-unet-v1.5-model-b2c4h64w64s77-float16-compute-and-inputs-layernorm.onnx",
+      path: "./demos/stable-diffusion-1.5/models"
+    },
+    {
+      url: "microsoft/stable-diffusion-v1.5-webnn/resolve/main/Stable-Diffusion-v1.5-vae-decoder-float16-fp32-instancenorm.onnx",
+      path: "./demos/stable-diffusion-1.5/models"
+    },
+    {
+      url: "microsoft/stable-diffusion-v1.5-webnn/resolve/main/safety-checker.onnx",
+      path: "./demos/stable-diffusion-1.5/models"
+    },
+    {
+      url: "microsoft/segment-anything-model-webnn/resolve/main/sam_vit_b_01ec64.encoder-fp16.onnx",
+      path: "./demos/segment-anything/models"
+    },
+    {
+      url: "microsoft/segment-anything-model-webnn/resolve/main/sam_vit_b_01ec64.decoder-fp16.onnx",
+      path: "./demos/segment-anything/models"
+    },
+    {
+      url: "microsoft/segment-anything-model-webnn/resolve/main/sam_vit_b-encoder-int8.onnx",
+      path: "./demos/segment-anything/models"
+    },
+    {
+      url: "microsoft/segment-anything-model-webnn/resolve/main/sam_vit_b-decoder-int8.onnx",
+      path: "./demos/segment-anything/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_kvcache_128_lm_fp16_layernorm.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_kvcache_128_lm_fp16_layernorm_4dmask.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_kvcache_128_lm_fp16_layernorm_gelu_4dmask.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_non_kvcache_lm_fp16_layernorm.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_non_kvcache_lm_fp16_layernorm_4dmask.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_decoder_static_non_kvcache_lm_fp16_layernorm_gelu_4dmask.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_encoder_lm_fp16_layernorm.onnx",
+      path: "./demos/whisper-base/models"
+    },
+    {
+      url: "microsoft/whisper-base-webnn/resolve/main/whisper_base_encoder_lm_fp16_layernorm_gelu.onnx",
+      path: "./demos/whisper-base/models"
+    }
+  ];
+
+  const downloadFile = async (url, outputPath, retries = 2) => {
+    // // Proxy configuration
+    // const proxyHost = '';
+    // const proxyPort = 33630;
+    // const proxyUrl = `http://${proxyHost}:${proxyPort}`;
+    // const proxyAgent = new HttpsProxyAgent(proxyUrl);
+
+    try {
+      // const res = await fetch(url, {agent: proxyAgent});
```

Comment:
To reduce the number of lines one has to uncomment/comment to toggle between proxy or not, can we just set this to the following (where proxyAgent = undefined in the non-proxy case, if that's allowed?):

```js
    // Proxy configuration
    const proxyHost = '';
    const proxyPort = 33630; // Special port used by xyz
    const proxyUrl = `http://${proxyHost}:${proxyPort}`;
    const proxyAgent = undefined;
    // Enable this line to use the proxy:
    // const proxyAgent = new HttpsProxyAgent(proxyUrl);

    try {
      const response = await fetch(url, {agent: proxyAgent});
...
```

Reply:
JS default function parameters allow named parameters to be initialized with default values if no value or undefined is passed. Use your code instead, thanks!